src/pages/core/architecture/events.mdx

```diff
@@ -0,0 +1,31 @@
+---
+tags: ["core", "architecture"]
+---
+
+# Events
+
+CosmWasm can emit
+[Cosmos Events](https://docs.cosmos.network/v0.50/learn/advanced/events). These
+events are stored in the block as metadata, allowing the contract to attach
+metadata to what exactly happened during execution.
+
+By default CosmWasm emits the `wasm` event to which you can add context like so:
+
+```rust filename="wasm_event.rs" template="core"
+let response: Response<Empty> = Response::new()
+    .add_attribute("custom_attribute", "value");
+```
+
```

Comment:
Some info that is missing here:
- whitespace is trimmed
- empy key (or whitespace only) is not allowed
-  keys cannot start with `_` (reserved for wasmd)